wled00/wled_server.cpp

```diff
@@ -443,16 +462,21 @@ void serveSettings(AsyncWebServerRequest* request, bool post)
       case 3: strcpy_P(s, PSTR("UI")); break;
       case 4: strcpy_P(s, PSTR("Sync")); break;
       case 5: strcpy_P(s, PSTR("Time")); break;
-      case 6: strcpy_P(s, PSTR("Security")); strcpy_P(s2, PSTR("Rebooting, please wait ~10 seconds...")); break;
+      case 6: strcpy_P(s, PSTR("Security")); if (doReboot) strcpy_P(s2, PSTR("Rebooting, please wait ~10 seconds...")); break;
       case 7: strcpy_P(s, PSTR("DMX")); break;
       case 8: strcpy_P(s, PSTR("Usermods")); break;
+      case 252: strcpy_P(s, correctPIN ? PSTR("PIN accepted") : PSTR("PIN rejected")); break;
     }
 
-    strcat_P(s, PSTR(" settings saved."));
+    if (subPage != 252) strcat_P(s, PSTR(" settings saved."));
+    else {
+      server.removeHandler(editHandler);
+      createEditHandler(correctPIN);
+    }
     if (!s2[0]) strcpy_P(s2, PSTR("Redirecting..."));
 
-    if (!doReboot) serveMessage(request, 200, s, s2, (subPage == 1 || subPage == 6) ? 129 : 1);
-    if (subPage == 6) doReboot = true;
+    serveMessage(request, 200, s, s2, (subPage == 1 || (subPage == 6 && doReboot)) ? 129 : (correctPIN ? 1 : 10));
```

Comment:
Putting a longer redirect time here unfortunately won't defeat a script (or browser back button).
We need to lock the set.cpp code from accepting new PINs for a few seconds (same for OTA lock now that reboot on Save is removed)

Reply:
The actual brute force protection is in line 448.